cost-model/src/cost_model.rs

```diff
@@ -42,70 +42,82 @@ enum SystemProgramAccountAllocation {
 }
 
 impl CostModel {
-    pub fn calculate_cost(
-        transaction: &SanitizedTransaction,
+    pub fn calculate_cost<'a>(
+        transaction: &'a SanitizedTransaction,
         feature_set: &FeatureSet,
-    ) -> TransactionCost {
+    ) -> TransactionCost<'a, SanitizedTransaction> {
         if transaction.is_simple_vote_transaction() {
-            TransactionCost::SimpleVote {
-                writable_accounts: Self::get_writable_accounts(transaction),
-            }
+            TransactionCost::SimpleVote { transaction }
         } else {
-            let mut tx_cost = UsageCostDetails::new_with_default_capacity();
-
-            Self::get_signature_cost(&mut tx_cost, transaction, feature_set);
-            Self::get_write_lock_cost(&mut tx_cost, transaction, feature_set);
-            Self::get_transaction_cost(&mut tx_cost, transaction, feature_set);
-            tx_cost.allocated_accounts_data_size =
+            let (signatures_count_detail, signature_cost) =
+                Self::get_signature_cost(transaction, feature_set);
+            let write_lock_cost = Self::get_write_lock_cost(transaction, feature_set);
+            let (programs_execution_cost, loaded_accounts_data_size_cost, data_bytes_cost) =
+                Self::get_transaction_cost(transaction, feature_set);
```

Comment:
It will have additional overhead if we iterate over instructions again, so would rather not add that.
This is not as much an issue w/ `SanitizedTransaction` but `TransactionView` we need to decode compressed u16 lengths for each ix then the overhead is higher.

Reply:
Yea, no problem to keep it on TODO for now.